src/NetscriptFunctions.ts

```diff
@@ -801,7 +801,10 @@ export const ns: InternalAPI<NSFull> = {
       const killByPid = typeof ident === "number";
       if (killByPid) {
         // Kill by pid
-        res = killWorkerScriptByPid(ident);
+        res = killWorkerScriptByPid(
+          ident,
+          `Script killed by script ${ctx.workerScript.name} with PID ${ctx.workerScript.pid}`,
```

Comment:
This log line is identical and duplicated, I'd construct it in `killWorkerScriptByPid` instead and pass `ctx.workerScript` as "killer".